website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+  verify_https_client    = true
+}
+```
+
+### Running with TLS
+
+Now that we have certificates generated and configuration for a client and
+server we can test our TLS-enabled cluster!
+
+In separate terminals start a server and client agent:
+
+```shell
+# In one terminal...
+nomad agent -config server1.hcl
+
+# ...and in another
+nomad agent -config client1.hcl
+```
+
+Finally in a third terminal test out `nomad node-status`:
+
+```text
+vagrant@nomad:~$ nomad node-status
+Error querying node status: Get http://127.0.0.1:4646/v1/nodes: malformed HTTP response "\x15\x03\x01\x00\x02\x02"
+```
+
+Oh no! That didn't work!
+
+Don't worry, the Nomad CLI just defaults to `http://...` instead of
+`https://...`. We can override this with an environment variable:
+
+```shell
+export NOMAD_ADDR=https://localhost:4646
+export NOMAD_CACERT=nomad-ca.pem
+export NOMAD_CLIENT_CERT=client.pem
+export NOMAD_CLIENT_KEY=client-key.pem
+```
+
+The `NOMAD_CACERT` also needs to be set so the CLI can verify it's talking to
+an actual Nomad node. Finally, `NOMAD_CLIENT_CERT` and `NOMAD_CLIENT_KEY` need
+to be set since we enabled `verify_https_client` above which prevents any
+access lacking a client certificate.
+
+Now the CLI works as expected:
+
+```text
+vagrant@nomad:~$ nomad node-status
+ID        DC   Name   Class   Drain  Status
+237cd4c5  dc1  nomad  <none>  false  ready
+
+vagrant@nomad:~$ nomad init
+Example job file written to example.nomad
+vagrant@nomad:~$ nomad run example.nomad
+==> Monitoring evaluation "e9970e1d"
+    Evaluation triggered by job "example"
+    Allocation "a1f6c3e7" created: node "237cd4c5", group "cache"
+    Evaluation within deployment: "080460ce"
+    Evaluation status changed: "pending" -> "complete"
+==> Evaluation "e9970e1d" finished with status "complete"
+```
+
+## Server Gossip
+
+We haven't quite completely secured Nomad's communications: Nomad server's
+gossip protocol uses a shared key instead of TLS for encryption. This
+encryption key must be added to every server's configuration using the
+[`encrypt`](/docs/agent/configuration/server.html#encrypt) parameter.
+
+As a convenience the Nomad CLI includes a `keygen` command for generating a new
+secure gossip encryption key:
+
+```text
+$ nomad keygen
+cg8StVXbQJ0gPvMd9o7yrg==
+```
+
+Put the generated key into each server's configuration file:
+
+```hcl
+server {
+    enabled = true
+
+    # Self-elect, should be 3 or 5 for production
+    bootstrap_expect = 1
+
+    # Encrypt gossip communication
+    encrypt = "cg8StVXbQJ0gPvMd9o7yrg=="
+}
+```
+
+## Switching an existing cluster to TLS
+
+Since Nomad does *not* use different ports for TLS and non-TLS communication,
+the use of TLS should be consistent across the cluster. Switching an existing
+cluster to use TLS everywhere is similar to upgrading between versions of
+Nomad.
+
+First make sure all of your nodes are ready to be switched:
+
+* Add the appropriate key and certificates to all nodes.
```

Comment:
Since these are ordered, we should use `1.` instead of bullets. You don't have to self-number. Use `1.` everywhere and markdown will do the right thing ™️